aspnetcore/data/ef-rp/concurrency.md

```diff
@@ -0,0 +1,333 @@
+---
+title: Razor Pages with EF Core - Concurrency - 8 of 8
+author: rick-anderson
+description: This tutorial shows how to handle conflicts when multiple users update the same entity at the same time.
+keywords: ASP.NET Core,Entity Framework Core,concurrency
+ms.author: riande
+manager: wpickett
+ms.date: 11/15/2017
+ms.topic: get-started-article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: data/ef-rp/concurrency
+---
+
+en-us/
+
+# Handling concurrency conflicts - EF Core with Razor Pages (8 of 8)
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT), [Tom Dykstra](https://github.com/tdykstra), and  [Jon P Smith](https://twitter.com/thereformedprog)
+
+[!INCLUDE[about the series](../../includes/RP-EF/intro.md)]
+
+This tutorial shows how to handle conflicts when multiple users update an entity concurrently (at the same time). If you run into problems you can't solve, download the [completed app for this stage](https://github.com/aspnet/Docs/tree/master/aspnetcore/data/ef-rp/intro/samples/StageSnapShots/cu-part8).
+
+## Concurrency conflicts
+
+A concurrency conflict occurs when:
+
+* A user navigates to the edit page for an entity.
+* Another user updates the same entity before the first user's change is written to the DB.
+
+If concurrency detection is not enabled, when concurrent updates occur:
+
+* The last update wins. That is, the last update values are saved to the DB.
+* The first of the current updates are lost. That's generally not what you want.
```

Comment:
> That's generally not what you want.

I currently disagree with this statement. I think we must recognize that concurrency is useful only in a subset of scenarios. When it matters, you can really use help from your persistence framework. I.e. concurrency control features can help make applications that handle multiple users almost as easy to code as single-user logic (and that is why we support it and why it is important to document how it works), but still we don't do any concurrency control by default.

Whether you care about concurrency is going to depend on whether there are rules that constrain what changes to the database are valid, e.g. if an application performs a debit transaction against a bank account, it matter that things balanced out at the end and that there is a non-negative balance at the end. If you validate that this is the case in the unit-of-work, then you can use concurrency to make sure no other debit transaction will be able to happen concurrently on the same account. But within the same application, you may not care if the description of the account is updated. @